File: Loenn/entities/cockatiel.lua
local utils = require("utils")

local cockatiel = {}
cockatiel.name = "JungleHelper/Cockatiel"
cockatiel.depth = -9999
cockatiel.placements = {
    {
        name = "default",
        data = {
            facingLeft = true,
            sprite = "",
        }
    }
}

function cockatiel.scale(room, entity)
    local scaleX = entity.facingLeft and 1 or -1
    return scaleX, 1
end

function cockatiel.selection(room, entity)
    return utils.rectangle(entity.x - 6, entity.y - 4, 12, 12)
end

cockatiel.offset = { 6, 4 }
cockatiel.texture = "JungleHelper/Cockatiel/idle00"

return cockatiel

File: Ahorn/entities/cockatiel.jl
module JungleHelperCockatiel

using ..Ahorn, Maple

@mapdef Entity "JungleHelper/Cockatiel" Cockatiel(x::Integer, y::Integer, facingLeft::Bool=true, sprite::String="")

const placements = Ahorn.PlacementDict(
    "Cockatiel (Jungle Helper)" => Ahorn.EntityPlacement(
        Cockatiel
    )
)

function Ahorn.selection(entity::Cockatiel)
    x, y = Ahorn.position(entity)
    return Ahorn.Rectangle(x - 6, y - 4, 12, 12)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Cockatiel, room::Maple.Room)
    scaleX = (get(entity.data, "facingLeft", false) ? 1 : -1)
    Ahorn.drawSprite(ctx, "JungleHelper/Cockatiel/idle00", 0, 2, sx = scaleX)
end

end
